### PR TITLE
Upgrade to postgres 10.15

### PR DESCRIPTION
**Pull request by bjgill**
Trello: https://trello.com/c/WdULXylT/2141-upgrade-postgresql

We need to upgrade to Postgres 12 by mid-February. We can't go directly, we need to go via Postgres 10 - https://docs.cloud.service.gov.uk/deploying_services/postgresql/#upgrade-to-postgresql-10.

Upgrade dmrunner to use postgres 10 so we can test locally. After pulling this commit, you will need to run `make data` to re-initialise the database.

### CODE REVIEW

**Comment by bjgill** [comment]
Run the functional tests locally. Smoke tests all OK. Some of the full suite failed - now to see whether that's a pre-existing condition...

**Comment by bjgill** [comment]
Yes - the same tests failed on Postgres 9 and 10 because I didn't have a Notify key. I think this is good to go.

File: config/docker-compose.yml [hunk 1]
```diff
@@ -8,14 +8,15 @@ volumes:
...
-    image: "postgres@sha256:f4603c7b8aaf418393edb8cd5e2d1abd91d686ab571302dc83f887ea4a56286b"
+    image: "postgres:10.15@sha256:95f79893ef4d769277f7687b03bb24fe77a49ad2ec1c269f89a0cf849d6f4c7b"                                                                   
     ports:
       - "5432:5432"
     volumes:
       - ./sql:/docker-entrypoint-initdb.d
       - postgres-data:/var/lib/postgresql/data
     environment:
       POSTGRES_USER: $USER
+      POSTGRES_HOST_AUTH_METHOD: trust
```
Comment (lfdebrux) [inline config/docker-compose.yml:19]: Why is this needed?

**Review by lfdebrux — APPROVED** [review]

File: config/docker-compose.yml (hunk 1 above)
Comment (bjgill) [reply config/docker-compose.yml:19]: Argh - meant to break this out into a separate commit, but forgot. It's due to a change in the Docker container that now requires us to be explicit that we don't want a root password: https://github.com/docker-library/postgres/pull/658

Comment (bjgill) [reply config/docker-compose.yml:19]: I've amended the commit description to include this. Will now merge when green
